ios/Event+JSON.swift

```diff
@@ -130,3 +130,140 @@ extension TimedEventType {
 extension PlayEvent: TimedEventType {}
 extension PausedEvent: TimedEventType {}
 extension PlayingEvent: TimedEventType {}
+
+extension CueEvent where Self: Event {
+    func toJSON() -> [AnyHashable: Any] {
+        // Initial JS event object.
+        var json: [AnyHashable: Any] = [
+            "name": name,
+            "timestamp": timestamp,
+            "html": html,
+            "text": text,
+            "start": startTime,
+            "end": endTime
+        ]
+
+        // Initial JS cue object.
+        var cueJson: [AnyHashable: Any] = [
+            "text": text,
+            "html": html,
+            "start": startTime,
+            "end": endTime,
+        ]
+
+        // Compute cue image base64 if available.
+        if let data = image?.pngData() {
+            // Source string should be compatible with RN's `Image` base64 data format.
+            let imageSrc = "data:image/png;base64,\(data.base64EncodedString())"
+            json["image"] = imageSrc
+            cueJson["image"] = imageSrc
+        }
+
+        // Define CEA-only positioning properties.
+        if let position = position {
+            cueJson["ceaPosition"] = [
```

Comment:
Yes, that's right for the `position` name, which just gets mapped to `ceaPosition`, but the properties `row` and `column` stay the same.

I decided to use a different name for `position` here because I noticed there're two different position-related properties for cues in iOS:

- [`position`](https://cdn.bitmovin.com/player/ios/3/docs/Classes/Cue.html#/c:@M@BitmovinPlayer@objc(cs)BMPCue(py)position) (which the docs state to be CEA-related)
- [`vtt`](https://cdn.bitmovin.com/player/ios/3/docs/Classes/Cue.html#/c:@M@BitmovinPlayer@objc(cs)BMPCue(py)vtt)

But when I was about to add support for Android cues, I noticed the SDK provides some properties which seem to be related to vtt positioning (there's no `position` prop like in iOS though):

- [`fractionalPosition`](https://cdn.bitmovin.com/player/android/3/docs/-player%20-module/com.bitmovin.player.api.media.subtitle/-cue/fractional-position.html)
- [`positionAnchor`](https://cdn.bitmovin.com/player/android/3/docs/-player%20-module/com.bitmovin.player.api.media.subtitle/-cue/position-anchor.html)
- [`line`](https://cdn.bitmovin.com/player/android/3/docs/-player%20-module/com.bitmovin.player.api.media.subtitle/-cue/line.html)
- [`lineType`](https://cdn.bitmovin.com/player/android/3/docs/-player%20-module/com.bitmovin.player.api.media.subtitle/-cue/line-type.html)
- [`lineAnchor`](https://cdn.bitmovin.com/player/android/3/docs/-player%20-module/com.bitmovin.player.api.media.subtitle/-cue/line-anchor.html)

So in order to find a common place between iOS and Android, I created two different keys in the RN cue which are `ceaPosition` and `vttPosition`. The first maps to iOS' `position` prop and gets `null` for Android, and the latter maps to `vtt` in iOS and the positioning props found in Android.

This way, we're able to have both information abstracted in the same cue object while expliciting stating what subtitle/CC format is associated with it.

Also, CEA-related positioning data seems to be missing from the Android SDK. Is that right @matamegger?

Reply:
If you think having both cue objects abstracted in a single interface from JS is a problem, I can split them up into something like `CueiOS` and `CueAndroid`. That's feasible too.

Reply:
We've agreed to drop cues support for now, see [commit](https://github.com/bitmovin/bitmovin-player-react-native/pull/14/commits/1599353362e30aa2471b60d7ddcff02d88bf8b2c).